src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -2,125 +2,115 @@
     }
 
-    /**
-     * Gets the relation data from the method in the model.
-     *
-     * @param ReflectionMethod $method
-     * @return array
-     */
-    public function inferFieldAttributesFromRelationship($method)
+    public function parseRelationFieldNamesFromHtml($fields)
     {
-        // get the parent of the last relation if using dot notation
-        // eg: user.account.address -> Return model for account and the relation address in account model.
-        $relationModel = $this->getRelationModel($method, -1);
-        $relatedMethod = Arr::last(explode('.', $method));
-
-        if ($relationModel != get_class($this->model)) {
-            $relationModel = new $relationModel();
-            if (method_exists($relationModel, $relatedMethod)) {
-                return $this->getFieldAttributesFromRelationship($relationModel, $relatedMethod);
+        foreach ($fields as &$field) {
+            if (isset($field['relation_type'])) {
+                $field['name'] = Arr::first(explode('.', $field['entity']));
             }
         }
-        if (method_exists($this->model, $relatedMethod)) {
-            return $this->getFieldAttributesFromRelationship($this->model, $relatedMethod);
-        }
 
-        return false;
+        return $fields;
     }
 
-    public function getFieldAttributesFromRelationship($model, $method)
+    public function setRelationHtmlName($relation_model, $entity)
     {
-        try {
-            $method = (new \ReflectionClass($model))->getMethod($method);
-
-            $relation = $method->invoke($model);
+        $entity_array = explode('.', $entity);
+        $name_string = '';
 
-            if ($relation instanceof Relation) {
-                $relationship['type'] = 'relationship';
-                $relationship['entity'] = $method->getName();
-                $relationship['relation_type'] = (new \ReflectionClass($relation))->getShortName();
-                $relationship['multiple'] = $this->relationAllowsMultiple($relationship['relation_type']);
-                $relationship['model'] = get_class($relation->getRelated());
-
-                if ($relationship['relation_type'] == 'BelongsTo' || $relationship['relation_type'] == 'HasOne') {
-                    $relationship['name'] = $relation->getForeignKeyName();
-                }
-
-                if ($relationship['relation_type'] == 'hasManyThrough' || $relationship['relation_type'] == 'BelongsToMany' || $relationship['relation_type'] == 'morphMany') {
-                    $relationship['pivot'] = true;
-                }
+        foreach ($entity_array as $key => $array_entity) {
+            $name_string .= ($key == 0) ? $array_entity : '['.$array_entity.']';
+        }
 
-                return $relationship;
-            }
+        return $name_string;
+    }
 
-            return false;
-        } catch (Exception $e) {
-            return false;
-        }
+    public function inferFieldTypeFromRelation($relation)
```

Comment:
Is this an empty method? 😄

Reply:
> Also, this breaks the relationship field for me, for 1-n relationship, where "name" has no dot notation:
> 
> https://drive.google.com/open?id=1IegvN7cE0_sHWbteaBnbUMQRkfqmZLGc
> 
> I don't think this is ready

I could not reproduce your error with icon. I am using the same branch as you: https://recordit.co/B7tpENb9JP

Reply:
Hmmm... I don't think so. I'm explicitly disabling InlineCreate. So I don't think it's getting used any more. It's just ajax=true, inline_create=false.

Reply:
Well, you are right, the same code is used in fetch and fetch_create that's why both are broken. 

Can you give it a go now? Both should be fixed, it was a matter of parsing the correct entity.

Reply:
Woohooo!! 🎈 🎉 🥇  Looks like it's working now - good job!